### PR TITLE
Fix build errors

### PR DESCRIPTION
**Pull request by mickaelandrieu**
# What is done in this contribution?

- [x] Sphinx updated from `1.2.0` to `1.5.3`
- [x] no longer use the outdated ubuntu package (updated README)
- [x] all warnings on docs build fixed
- [x] introduced Travis configuration (be able to refuse any contribution that creates warnings)

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -33,8 +33,6 @@
 lexers['php'] = PhpLexer(startinline=True)
 lexers['php-annotations'] = PhpLexer(startinline=True)
 
-primary_domain = 'php'
```
Comment (mickaelandrieu) [inline conf.py]: we should use sphinx 1.5.* instead of 1.2 (I'll update installation docs of sphinx)

**Comment by nidup** [comment]
Cool PR, future will be better with that check \o/

File: index.rst [hunk 2]
```diff
@@ -19,4 +20,3 @@ efficient answers to their multichannel needs.
     cookbook/index
     contributing/index
     troubleshooting/index
-    styleguide/index
```
Comment (pierallard) [inline index.rst]: Nope

**Comment by mickaelandrieu** [comment]
@pierallard how? Afaik, Sphinx is not installed on staging :/

**Comment by mickaelandrieu** [comment]
Sounds OK to me on staging => GTM

ok

**Comment by mickaelandrieu** [comment]
@akeneo/core-team I need someone to register this repository on travis please 👍

**Comment by nidup** [comment]
@mickaelandrieu done!

**Pull request merged**